File: 739-DailyTemperatures/739-DailyTemperatures.cpp
// Last updated: 3/26/2026, 1:26:27 PM
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n = temperatures.size();
        vector<int> ans(n,0);
        stack<int> st;
        for(int i =n-1;i>=0;i--){
            while(!st.empty() && temperatures[st.top()]<=temperatures[i])st.pop();
            

            if(!st.empty())ans[i] = st.top()-i;

        st.push(i);
        }

        return ans;
        
    }
};

File: 739-DailyTemperatures/739-DailyTemperatures.java
// Last updated: 3/26/2026, 1:26:21 PM
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>();
        int n = temperatures.length;
        int[] answer = new int[n];

        for(int i = n-1;i>=0;i--){
            while(!stack.isEmpty() && temperatures[stack.peek()]<=temperatures[i]){
            stack.pop();
            }
            if(!stack.isEmpty()){
                answer[i] = stack.peek() - i; 
            }
            stack.push(i);

        }
        return answer;
    }
}